### PR TITLE
Allow using less than full balance for partially fillable orders

### PR DESCRIPTION
**Pull request by vkgnosis**
Related to https://github.com/cowprotocol/services/issues/1378 .

The full implementation of that issue is more complicated than expected. Instead, this PR implements a simple version of it that brings the main benefit. The main benefit is that a partially fillable order, for which the user doesn't have full sell balance, can still be settled.

This simple version continue to let the autopilot make the choice of what partially fillable orders to include and forwards the balance for the order in the auction data so that it can be used by the driver.

This is not yet (properly) implemented for the colocation code because we don't pass the available balance along there.

### Test Plan

adapted unit tests and new e2e test

### CODE REVIEW

File: crates/solver/src/liquidity/order_converter.rs [hunk 1]
```diff
@@ -58,12 +54,49 @@ impl OrderConverter {
...
+                .checked_add(remaining.remaining(order.data.fee_amount)?)
+                .context("partially fillable")?;
+            let have = order
+                .metadata
+                .partially_fillable_balance
+                .context("no balance for partially fillable order")?;
+            tracing::trace!(%need, %have, "partially fillable order conversion");
+            if have < need {
+                solver_fee = solver_fee
+                    .checked_mul(have)
```
Comment (MartinquaXD) [inline crates/solver/src/liquidity/order_converter.rs:77]: You could create another `Remaining` instance from `have` and `need` to compute the adjusted amounts. Might look a little nicer.

**Review by MartinquaXD — APPROVED** [review]
Don't know what the complications are with the proper solution but this seems like a reasonable intermediate solution.

File: crates/solver/src/liquidity/order_converter.rs (hunk 1 above)
Comment (vkgnosis) [reply crates/solver/src/liquidity/order_converter.rs:77]: The Remaining struct does a similar calculation but it was made for the order executed amounts and for example numberator, denominator are private. I didn't want to generalise it for this so I repeated the calculation.

File: crates/solver/src/liquidity/order_converter.rs [hunk 2]
```diff
@@ -58,12 +54,49 @@ impl OrderConverter {
...
+        let mut solver_fee = remaining.remaining(order.metadata.solver_fee)?;
+        let mut sell_amount = remaining.remaining(sell_amount)?;
+        let mut buy_amount = remaining.remaining(order.data.buy_amount)?;
+
+        // Partially fillable orders are included in the auction when there is at least
+        // 1 atom balance available.
+        if order.data.partially_fillable {
+            let need = sell_amount
+                .checked_add(remaining.remaining(order.data.fee_amount)?)
+                .context("partially fillable")?;
```
[the remark this answers is not in the document]
Comment (vkgnosis) [reply crates/solver/src/liquidity/order_converter.rs:69]: Will make it a bit nicer.

**Comment by vkgnosis** [comment]
I'm merging this because it makes testing easier so you can place orders that must be partial matched because missing balance. Reviews still appreciated.

**Pull request merged**